applicationset/generators/git.go

```diff
@@ -60,20 +60,31 @@ func (g *GitGenerator) GenerateParams(appSetGenerator *argoprojiov1alpha1.Applic
 	noRevisionCache := appSet.RefreshRequired()
 
 	var project string
+	isProjectTemplated := false
+
 	if strings.Contains(appSet.Spec.Template.Spec.Project, "{{") {
-		project = appSetGenerator.Git.Template.Spec.Project
+		// When the project field is templated, the contents of the git repo are required to run the git generator and get the templated value,
+		// but git generator cannot be called without verifying the commit signature.
+		// In this case, we skip the signature verification.
+		if strings.Contains(appSet.Spec.Template.Spec.Project, "{{") {
+			isProjectTemplated = true
+		} else {
+			project = appSetGenerator.Git.Template.Spec.Project
+		}
```

Comment:
I might be reading wrong, but isn't this internal condition redundant?

Reply:
You are right, simplified the code logic.